### PR TITLE
Add predicate attribute unit test.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

File: searchlib/src/tests/attribute/predicate_attribute/predicate_attribute_test.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+class PredicateAttributeTest : public ::testing::Test
+{
+protected:
+    PredicateAttributeTest();
+    ~PredicateAttributeTest();
+    void SetUp() override;
+    void TearDown() override;
+};
+
+PredicateAttributeTest::PredicateAttributeTest()
```
Comment (baldersheim) [inline searchlib/src/tests/attribute/predicate_attribute/predicate_attribute_test.cpp:77]: = default ...

File: searchlib/src/tests/attribute/predicate_attribute/predicate_attribute_test.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    : ::testing::Test()
+{
+}
+
+PredicateAttributeTest::~PredicateAttributeTest()
```
Comment (baldersheim) [inline searchlib/src/tests/attribute/predicate_attribute/predicate_attribute_test.cpp:82]: = default

File: searchlib/src/tests/attribute/predicate_attribute/predicate_attribute_test.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    return os.str();
+}
+
+}
+
+class PredicateAttributeTest : public ::testing::Test
+{
+protected:
+    PredicateAttributeTest();
+    ~PredicateAttributeTest();
```
Comment (baldersheim) [inline searchlib/src/tests/attribute/predicate_attribute/predicate_attribute_test.cpp:72]: override ?

File: searchlib/src/tests/attribute/predicate_attribute/predicate_attribute_test.cpp [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    search::attribute::PredicateParams predicateParams;
+    predicateParams.setArity(arity);
+    ret.setPredicateParams(predicateParams);
+    return ret;
+}
+
+std::shared_ptr<AttributeVector>
+make_attribute(vespalib::stringref name, const Config& cfg, bool setup)
+{
+    auto attribute = AttributeFactory::
```
Comment (baldersheim) [inline searchlib/src/tests/attribute/predicate_attribute/predicate_attribute_test.cpp:50]: avoid line break

**Review by baldersheim — DISMISSED** [review]
Some minor comments

**Review by baldersheim — APPROVED** [review]

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**